account-specific.tf

```diff
@@ -73,10 +75,12 @@ data "aws_iam_policy_document" "account_specific_policy" {
 }
 
 resource "aws_iam_policy" "account_state_policy" {
-  count       = length(var.account_arns)
-  name        = "${replace(var.account_arns[count.index], local.account_from_arn, "$5")}-terraform-state-assumed-policy"
+  count = length(var.account_arns)
+  name  = "${replace(var.account_arns[count.index], local.account_from_arn, "$5")}-terraform-state-assumed-policy"
+  # tflint-ignore: terraform_deprecated_index ignored because terraform will threat it as list
   description = "Policy given upon role assumption of ${replace(var.account_arns[count.index], local.account_from_arn, "$5")}-terraform-state role"
-  policy      = data.aws_iam_policy_document.account_specific_policy[*].json[count.index]
+  # tflint-ignore: terraform_deprecated_index ignored because terraform will threat it as list
+  policy = data.aws_iam_policy_document.account_specific_policy.*.json[count.index]
```

Comment:
Similar to the above - do we want to go backwards to the legacy splat function, as opposed to the modern version?  Is it better-suited for this purpose?

Reply:
This is just for ignoring and "error" because tflint is thinking it is a list, but the code is OK and unchanged

Reply:
Look at line 79 and line 83. It appears there is definitely a change, from _policy[*].json to _policy.*.json...which, according to the official documentation, changes the behavior of the operator.  Am I reading that incorrectly?